sdk/eventhub/Azure.Messaging.EventHubs/tests/Processor/EventProcessorLiveTests.cs

```diff
@@ -0,0 +1,907 @@
+                            client,
+                            onProcessEvents: (partitionContext, checkpointManager, events, cancellationToken) =>
+                            {
+                                if (events.Count() > 0)
+                                {
+                                    allReceivedEvents.AddOrUpdate
+                                    (
+                                        partitionContext.PartitionId,
+                                        partitionId => new List<EventData>(events),
+                                        (partitionId, list) =>
+                                        {
+                                            list.AddRange(events);
+                                            return list;
+                                        }
+                                    );
+                                }
+                            }
+                        );
+
+                    hub.AddEventProcessors(1);
+
+                    // Send some events.
+
+                    var partitionIds = await client.GetPartitionIdsAsync();
+                    var expectedEvents = new Dictionary<string, List<EventData>>();
+
+                    foreach (var partitionId in partitionIds)
+                    {
+                        // Send a similar set of events for every partition.
+
+                        expectedEvents[partitionId] = new List<EventData>
+                        {
+                            new EventData(Encoding.UTF8.GetBytes($"{ partitionId }: event processor tests are so long.")),
+                            new EventData(Encoding.UTF8.GetBytes($"{ partitionId }: there are so many of them.")),
+                            new EventData(Encoding.UTF8.GetBytes($"{ partitionId }: will they ever end?")),
+                            new EventData(Encoding.UTF8.GetBytes($"{ partitionId }: let's add a few more messages.")),
+                            new EventData(Encoding.UTF8.GetBytes($"{ partitionId }: this is a monologue.")),
+                            new EventData(Encoding.UTF8.GetBytes($"{ partitionId }: loneliness is what I feel.")),
+                            new EventData(Encoding.UTF8.GetBytes($"{ partitionId }: the end has come."))
+                        };
+
+                        await using (var producer = client.CreateProducer(new EventHubProducerOptions { PartitionId = partitionId }))
+                        {
+                            await producer.SendAsync(expectedEvents[partitionId]);
+                        }
+                    }
+
+                    // Start the event processors.
+
+                    await hub.StartAllAsync();
+
+                    // Make sure the event processors have enough time to stabilize and receive events.
+                    // TODO: we'll probably need to extend this delay once load balancing is implemented.
+
+                    await Task.Delay(5000);
```

Comment:
Same here as the others.   Since you're round-tripping events and know that you have a set number expected, maybe we can do something to intercept as they're received and make a decision when they're all accounted for,  falling back to a "well, if it's been more than XX time, they're not coming and we've got a problem" ...

Reply:
Again, not necessary for this pass, but we'll definitely want to revisit.

Reply:
Just opened an issue: #7375